text_generation/causal_lm/cpp/CMakeLists.txt

```diff
@@ -5,26 +5,45 @@ cmake_minimum_required(VERSION 3.15)
+
+set(TARGET_NAME beam_search_causal_lm)
+add_executable(${TARGET_NAME} beam_search_causal_lm.cpp)
+target_include_directories(${TARGET_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
+target_compile_definitions(${TARGET_NAME} PRIVATE OPENVINO_TOKENIZERS_PATH=\"$<TARGET_FILE:openvino_tokenizers>\")
 find_package(OpenVINO REQUIRED COMPONENTS Runtime)
-target_link_libraries(greedy_causal_lm PRIVATE openvino::runtime)
-set_target_properties(greedy_causal_lm PROPERTIES CXX_STANDARD 17)
-set_target_properties(greedy_causal_lm PROPERTIES CXX_STANDARD_REQUIRED ON)
+target_link_libraries(${TARGET_NAME} PRIVATE openvino::runtime)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD 17)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD_REQUIRED ON)
 
-add_executable(beam_search_causal_lm beam_search_causal_lm.cpp)
-target_compile_definitions(beam_search_causal_lm PRIVATE OPENVINO_TOKENIZERS_PATH=\"$<TARGET_FILE:openvino_tokenizers>\")
-target_include_directories(beam_search_causal_lm PRIVATE ./)
+set(TARGET_NAME speculative_decoding_lm)
+add_executable(${TARGET_NAME} speculative_decoding_lm.cpp)
+target_include_directories(${TARGET_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
+target_compile_definitions(${TARGET_NAME} PRIVATE OPENVINO_TOKENIZERS_PATH=\"$<TARGET_FILE:openvino_tokenizers>\")
 find_package(OpenVINO REQUIRED COMPONENTS Runtime)
-target_link_libraries(beam_search_causal_lm PRIVATE openvino::runtime)
-set_target_properties(beam_search_causal_lm PROPERTIES CXX_STANDARD 17)
-set_target_properties(beam_search_causal_lm PROPERTIES CXX_STANDARD_REQUIRED ON)
+target_link_libraries(${TARGET_NAME} PRIVATE openvino::runtime)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD 17)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD_REQUIRED ON)
 
-add_executable(speculative_decoding_lm speculative_decoding_lm.cpp)
-target_compile_definitions(speculative_decoding_lm PRIVATE OPENVINO_TOKENIZERS_PATH=\"$<TARGET_FILE:openvino_tokenizers>\")
-target_include_directories(speculative_decoding_lm PRIVATE ./)
+set(TARGET_NAME generate_sample)
+add_executable(${TARGET_NAME} generate_pipeline/generate_sample.cpp)
+target_include_directories(${TARGET_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
+target_compile_definitions(${TARGET_NAME} PRIVATE OPENVINO_TOKENIZERS_PATH=\"$<TARGET_FILE:openvino_tokenizers>\")
 find_package(OpenVINO REQUIRED COMPONENTS Runtime)
-target_link_libraries(speculative_decoding_lm PRIVATE openvino::runtime)
-set_target_properties(speculative_decoding_lm PROPERTIES CXX_STANDARD 17)
-set_target_properties(speculative_decoding_lm PROPERTIES CXX_STANDARD_REQUIRED ON)
+target_link_libraries(${TARGET_NAME} PRIVATE openvino::runtime)
+target_link_libraries(${TARGET_NAME} PRIVATE nlohmann_json::nlohmann_json)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD 17)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD_REQUIRED ON)
+
+target_link_libraries(${TARGET_NAME} PRIVATE stdc++fs)
```

Comment:
std::filesystem is part of c++17, while compilers support differently: https://github.com/cnjinhao/nana/issues/423
- Windows MSVC: does not accept stdc++fs option, which cause build link error on windows
- Linux gcc: gcc 9.1 support it natively w/o `stdc++fs`, while for older gcc version, `stdc++fs` required

Should we dispatch for different compiler version? 
BTW, do we really need to introduce filesystem, it seems only check if generation_config.json exist.

Reply:
Removed experimental `stdc++fs`, i needed it when i was working on an older laptop with old gcc which supported only experimental fs. Since we use c++17 now this should work without any experimental namespaces and `stdc++fs`.